src/Api/Billing/Controllers/ProviderBillingController.cs

```diff
@@ -13,59 +20,195 @@ namespace Bit.Api.Billing.Controllers;
     ICurrentContext currentContext,
     IFeatureService featureService,
-    IProviderBillingService providerBillingService) : Controller
+    IProviderBillingService providerBillingService,
+    IProviderRepository providerRepository,
+    IStripeAdapter stripeAdapter,
+    ISubscriberService subscriberService) : Controller
 {
-    [HttpGet("subscription")]
-    public async Task<IResult> GetSubscriptionAsync([FromRoute] Guid providerId)
+    [HttpGet("payment-information")]
+    public async Task<IResult> GetPaymentInformationAsync([FromRoute] Guid providerId)
     {
-        if (!featureService.IsEnabled(FeatureFlagKeys.EnableConsolidatedBilling))
+        var (provider, result) = await GetAuthorizedBillableProviderOrResultAsync(providerId);
+
+        if (provider == null)
+        {
+            return result;
+        }
+
+        var paymentInformation = await subscriberService.GetPaymentInformation(provider);
+
+        if (paymentInformation == null)
         {
             return TypedResults.NotFound();
         }
 
-        if (!currentContext.ProviderProviderAdmin(providerId))
+        var response = PaymentInformationResponse.From(paymentInformation);
+
+        return TypedResults.Ok(response);
+    }
+
+    [HttpGet("payment-method")]
+    public async Task<IResult> GetPaymentMethodAsync([FromRoute] Guid providerId)
+    {
+        var (provider, result) = await GetAuthorizedBillableProviderOrResultAsync(providerId);
+
+        if (provider == null)
         {
-            return TypedResults.Unauthorized();
+            return result;
         }
 
-        var providerSubscriptionDTO = await providerBillingService.GetSubscriptionDTO(providerId);
+        var maskedPaymentMethod = await subscriberService.GetPaymentMethod(provider);
 
-        if (providerSubscriptionDTO == null)
+        if (maskedPaymentMethod == null)
         {
             return TypedResults.NotFound();
         }
 
-        var (providerPlans, subscription) = providerSubscriptionDTO;
+        var response = MaskedPaymentMethodResponse.From(maskedPaymentMethod);
+
+        return TypedResults.Ok(response);
+    }
+
+    [HttpPut("payment-method")]
+    public async Task<IResult> UpdatePaymentMethodAsync(
+        [FromRoute] Guid providerId,
+        [FromBody] TokenizedPaymentMethodRequestBody requestBody)
+    {
+        var (provider, result) = await GetAuthorizedBillableProviderOrResultAsync(providerId);
+
+        if (provider == null)
+        {
+            return result;
+        }
+
+        var tokenizedPaymentMethod = new TokenizedPaymentMethodDTO(
+            requestBody.Type,
+            requestBody.Token);
 
-        var providerSubscriptionResponse = ProviderSubscriptionResponse.From(providerPlans, subscription);
+        await subscriberService.UpdatePaymentMethod(provider, tokenizedPaymentMethod);
 
-        return TypedResults.Ok(providerSubscriptionResponse);
+        // TODO: Do we need to try and pay the outstanding invoices here?
```

Comment:
⛏️ We should generally avoid todos when possible. If you can get clarification from Clay or Priya, create a ticket off this so that you can remove the comment

Reply:
Opened this ticket to address: https://bitwarden.atlassian.net/browse/AC-2746